qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -329,19 +329,19 @@ def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) ->
             rtol = self.rtol
 
         label_list, indexes = np.unique(self._labels, return_inverse=True, axis=0)
-        coeff_list = np.zeros(len(self._coeffs))
```

Comment:
I guess another way might have been to set the data type on the np.zeros array as complex. But then it seems the coeff is can be passed in as a Union of int, float and complex (typehint of  complex implies this) and internally its left as the original type, so this would have made them all complex which perhaps was not desired.

Reply:
If all original coefficients are `List[float]` or `List[int]`, I thought  we need to make a zero array of float or int. Otherwise, the resulting coefficients are converted into `List[complex]` even if there is no complex values. If you are ok with it, I can change it as `np.zeros(..., dtype=complex)`. https://github.com/Qiskit/qiskit-nature/blob/52b0861d686ca5767626b809e7463257ae1bf95c/qiskit_nature/operators/second_quantization/fermionic_op.py#L116

Reply:
Yes, the internal keeps the original types float or int. So, I made this code to keep the original type. If we can convert the internal coeffs into complex, we can simply change this line as `np.zeros(..., dtype=complex)`.